synapse/handlers/read_marker.py

```diff
@@ -44,6 +44,9 @@ async def received_client_read_marker(
         """
 
         with await self.read_marker_linearizer.queue((room_id, user_id)):
+            if not self.hs.should_send_read_receipts():
+                return
+
```

Comment:
this one seems to be blocking requests from the local user to send receipts. As noted on https://github.com/matrix-org/synapse/issues/9253, that's risky - can you clarify the intention here?

(also, this can safely move outside the `read_marker_linearizer` block)

Reply:
Yep, shouldn't have put this here.